File: apps/android/feature/canvas/domain/src/main/java/com/scribblefit/feature/canvas/domain/repository/CanvasRepository.kt
package com.scribblefit.feature.canvas.domain.repository

import com.scribblefit.feature.canvas.domain.model.FeedItem
import kotlinx.coroutines.flow.Flow

interface CanvasRepository {
    fun getFeed(): Flow<List<FeedItem>>
    suspend fun addScribble(rawText: String)
    suspend fun retryScribble(id: String)
    suspend fun addConfirmation(item: FeedItem.Confirmation)
    suspend fun addInsight(item: FeedItem.Insight)
    suspend fun removeFeedItem(id: String)
    suspend fun clearFeed()
}

File: apps/ios/ScribbleFit/ScribbleFit/Features/Canvas/Domain/Repository/CanvasRepository.swift
import Foundation
import Combine

public protocol CanvasRepository: Sendable {
    /**
     * Observes the conversational feed for the current active session.
     */
    func getFeed() -> AnyPublisher<[FeedItem], Never>

    /**
     * Adds a raw user entry to the persistent sync queue and feed.
     */
    func addScribble(rawText: String) async throws

    /**
     * Retries a failed parsing attempt.
     */
    func retryScribble(id: String) async throws

    /**
     * Adds a structured confirmation card to the feed.
     */
    func addConfirmation(item: ConfirmationItem) async throws

    /**
     * Adds a micro-insight bubble to the feed.
     */
    func addInsight(item: InsightItem) async throws

    /**
     * Removes a specific item from the conversational feed.
     */
    func removeFeedItem(id: String) async throws

    /**
     * Clears the current canvas feed.
     */
    func clearFeed() async throws
}
